ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -209,8 +208,9 @@ private void cleanupExpiredInflightRequests() {
       DeleteRequestInfo deleteRequestInfo = itr.next().getValue();
       if (time.milliseconds() - deleteRequestInfo.startTimeMs > routerConfig.routerRequestTimeoutMs) {
         itr.remove();
-        responseHandler
-            .onRequestResponseException(deleteRequestInfo.replica, new IOException("Timed out waiting for a response"));
+        // Do not notify this as a failure to the response handler, as this timeout could simply be due to
```

Comment:
are you going to have a test to check that the failure detector is not notified (like you do in some other cases)? Or is it unnecessary?